rst_files/lucas_model.rst

```diff
@@ -414,12 +414,109 @@ Let's try this when :math:`\ln y_{t+1} = \alpha \ln y_t + \sigma \epsilon_{t+1}`
 
 Utility will take the isoelastic form :math:`u(c) = c^{1-\gamma}/(1-\gamma)`, where :math:`\gamma > 0` is the coefficient of relative risk aversion
 
-Some code to implement the iterative computational procedure can be found in
-`lucastree.jl <https://github.com/QuantEcon/QuantEcon.lectures.code/blob/master/lucas_model/lucastree.jl>`__
+Some code to implement the iterative computational procedure can be found below:
 
-We repeat it here for convenience
+.. code-block:: julia
+
+    #=
+
+    @authors : Spencer Lyon <spencer.lyon@nyu.edu>, John Stachurski
+
+    =#
+
+    using QuantEcon, Distributions
+
+    """
+    The Lucas asset pricing model --- parameters and grid data
+    """
+    struct LucasTree{TF<:AbstractFloat}
+        γ::TF                 # coefficient of risk aversion
+        β::TF                 # Discount factor in (0, 1)
+        α::TF                 # Correlation coefficient in the shock process
+        σ::TF                 # Volatility of shock process
+        ϕ::Distribution       # Distribution for shock process
+        grid::AbstractRange   # Grid of points on which to evaluate prices
+        shocks::Vector{TF}    # Draws of the shock
+        h::Vector{TF}         # The h function represented as a vector
+    end
+
+    """
+    Constructor for the Lucas asset pricing model
+    """
+    function LucasTree(;γ::AbstractFloat=2.0,
+                    β::AbstractFloat=0.95,
+                    α::AbstractFloat=0.9,
+                    σ::AbstractFloat=0.1,
+                    grid_size::Integer=100)
+
+        ϕ = LogNormal(0.0, σ)
+        shocks = rand(ϕ, 500)
+
+        # == build a grid with mass around stationary distribution == #
+        ssd = σ / sqrt(1 - α^2)
+        grid_min, grid_max = exp(-4 * ssd), exp(4 * ssd)
+        grid = range(grid_min, stop = grid_max, length = grid_size)
+        
+        # == set h(y) = β * int u'(G(y,z)) G(y,z) ϕ(dz) == #
+        h = similar(grid)
+        for (i, y) in enumerate(grid)
+            h[i] = β * mean((y^α .* shocks).^(1 - γ))
```

Comment:
Shouldn't this have `mean(x, dims = dims)`?

Reply:
Julia didn't complain but let me take a look.

Reply:
I think is fine, but didn't see a `using Statistics`.

Reply:
This collection is one-dimensional, so it doesn't make a semantic difference:

```
julia> mean([1, 2, 3, 4])
2.5

julia> mean([1, 2, 3, 4]')
2.5

julia> mean([1 2 3 4])
2.5
```

What this should have, though, is `using Statistics`. But we escaped, probably because `Distributions` exports its own version of `mean` that extends the base one.

Reply:
I guess it should be fine in this context.